src/hardware.md

```diff
@@ -37,3 +63,271 @@ We can write import scripts to handle virtually any "open" file format that cont
+| `rmf_fleet_msgs/FleetState`         | `/fleet_states`         | Provides RMF with fleet and mobile robot status updates, including the their locations, as well as their current path and task information. |
+
+(** These messages and topics are referenced from the code-base, it might not be the most up-to-date)
+
+The block diagram below shows an example implementation using a fleet of MiR100™ mobile robot, with a Full Control Fleet Adapter. On one end the Fleet Driver uses the ROS2 message and topic interfaces to interact with the Fleet Adapter, while on the other end it uses the REST API that comes with the MiR100™ robots to communicate and control each mobile robot.
+
+<img src="images/hardware_mir_fleet_driver_blocks.png">
+
+If however the user wishes to add more complexity to his or her fleet's behavior, the Fleet Adapter and Fleet Drivers can be implemented from scratch using the API provided in the `rmf_core` repository. In this scenario, the user can opt for implementing both the adapter and driver into a single module, without the need to use ROS2 topic and messages between them.
+
+### Standalone Mobile Robot
+
+In the event that the user wishes to integrate a standalone mobile robot which doesn't come with its own fleet management system, the open source fleet management system `free_fleet` could be used. 
+
+`free_fleet` can be split into a client and a server. The client is to be run on each of these standalone mobile robots alongside their navigational software, and is intended to have direct control over the mobile robot, while at the same time is able to monitor its status to be reported back to the server. The client's base implementation is designed to allow interaction with different configurations of mobile robots, while at the same time all report to the same server. This way, users are able to use `free_fleet` to manage a heterogenrous fleet of robots, each using different distributions of ROS, versions of ROS, navigation software, or onboard communication protocols.
+
+The server is run on a central computer, consolidates the incoming status updates from each client to be either visualized using the developer UI, or relayed upstream to RMF. The server also relays commands from the user via the UI or RMF, down to the clients to be executed. Each server can work with multiple clients at a time, hence it serves the role as a fleet management system. The server can be implemented and used as its own fleet management system, or work with larger systems like RMF, bridging the gap between each mobile robot's API and RMF's API and interface.
+
+The communication between the server and clients is implemented using `CycloneDDS`, therefore we are not concerned if the mobile robot or central computer is running different versions of ROS.
```

Comment:
Excuse my ignorance but is this correct? I thought we had Fast RTPS running on RMF? Should we specify the implementation of DDS or just specify that it is "implemented using DDS" instead of a specific brand/implementation?

Reply:
Oh, I meant to refer to the communication between the `free_fleet` client and `free_fleet` server. The ROS 2 `free_fleet` server then communicates with RMF using the default middleware which is `FastRTPS` at the moment